src/sign-in-windows-forms/MainWindow.cs

```diff
@@ -0,0 +1,116 @@
+using Microsoft.Identity.Client;
+using System.Net.Http.Headers;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+
+namespace MsalExample
+{
+    public partial class MainWindow : Form
+    {
+        private readonly HttpClient httpClient = new();
+
+        // Generally, your MSAL client will have a lifecycle that matches the lifecycle
+        // of the user's session in the application. In this sample, the lifecycle of the
+        // MSAL client to the lifecycle of this form.
+        private readonly IPublicClientApplication msalPublicClientApp;
+
+        public MainWindow()
+        {
+            InitializeComponent();
+
+            // Configure your public client application
+            msalPublicClientApp = PublicClientApplicationBuilder
+                .CreateWithApplicationOptions(new PublicClientApplicationOptions
+                {
+                    // 'Tenant ID' of your Azure AD instance - this value is a GUID
+                    TenantId = "",
+
+                    // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+                    ClientId = ""
+                })
+                .WithDefaultRedirectUri() // http://localhost
+                .Build();
+        }
+
+        /// <summary>
+        /// Handle the "Sign In" button click. This will acquire an access token scoped to
+        /// Microsoft Graph, either from the cache or from an interactive session. It will
+        /// then use that access token in an HTTP request to Microsoft Graph and display
+        /// the results.
+        /// </summary>
+        private async void SignInButton_Click(object sender, EventArgs e)
+        {
+            AuthenticationResult? msalAuthenticationResult = null;
+
+            // Acquire a cached access token for Microsoft Graph if one is available from a prior
+            // execution of this process.

```

Comment:
This _process_ or this _event handler_ (or method)? When I see "process" I think the `System.Diagnostics.Process`. If it's truly a prior execution of the process, consider instead "prior execution of the **application**," but I think that would require token serialization and a persistent cache, yeah?

Reply:
prior execution of the auth flow (which I was calling a process).  Reworded for added clarity.